modules/iam-role-for-service-accounts-eks/policies.tf

```
@@ -422,6 +422,50 @@ resource "aws_iam_role_policy_attachment" "efs_csi" {
   policy_arn = aws_iam_policy.efs_csi[0].arn
 }
 
+################################################################################
+# S3 CSI Driver Policy
+################################################################################
+
+#https://github.com/awslabs/mountpoint-s3/blob/main/doc/CONFIGURATION.md#iam-permissions
+data "aws_iam_policy_document" "s3_csi" {
+  count = var.create_role && var.attach_s3_csi_policy ? 1 : 0
+
+  statement {
+    sid       = "MountpointFullBucketAccess"
+    actions   = ["s3:ListBucket"]
+    resources = var.s3_csi_bucket_arns
```

to be more cross partition friendly

```suggestion
    resources = coalescelist(var.s3_csi_bucket_arns, ["arn:${local.partition}:s3:::*"])
```